test/pjrt/test_collective_ops_tpu.py

```diff
@@ -132,5 +139,126 @@ def test_all_to_all(self, pin_layout):
                                              list(range(world_size))]])
 
 
+# Test for collective ops from torch.distributed
+class TestDistCollectiveOpsTpu(parameterized.TestCase):
+
+  # TODO(zpcore): fix the openxla dynamo issue for dist.all_reduce, dist.all_gather
+  @staticmethod
+  def my_compiler(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]):
+    return gm.forward
+
+  @staticmethod
+  def _all_reduce(use_dynamo: bool):
+    met.clear_all()
+
+    def _dist_all_reduce(input):
+      dist.all_reduce(input, dist.ReduceOp.SUM)
+      return input
+
+    dist.init_process_group("xla", init_method='xla://')
+    device = xm.xla_device()
+    ordinal = torch.tensor([xr.global_ordinal()],
+                           dtype=torch.float,
+                           device=device)
+    input = ordinal
+
+    f = torch.compile(
+        _dist_all_reduce, backend=TestDistCollectiveOpsTpu.my_compiler
+    ) if use_dynamo else _dist_all_reduce
+    f(input)
+    torch_xla.sync()
+    if not use_dynamo:
+      assert 'xla::AllReduceInPlace' in met.counter_names(
+      ) or 'xla::AllReduce' in met.counter_names()
+    else:
+      assert 'xla::all_reduce' in met.counter_names()
+    return input.cpu()
+
+  @staticmethod
+  def _all_gather_into_tensor(use_dynamo: bool):
+    met.clear_all()
+
+    def _dist_all_gather_into_tensor(output, input):
+      dist.all_gather_into_tensor(output_tensor, input, None)
+      return output_tensor
+
+    dist.init_process_group("xla", init_method='xla://')
+    device = xm.xla_device()
+    ordinal = torch.tensor([xr.global_ordinal()],
+                           dtype=torch.float,
+                           device=device)
+    input = ordinal
+    output_tensor = torch.empty((1, xr.world_size()), device=device)
+    f = torch.compile(
+        _dist_all_gather_into_tensor,
+        backend=TestDistCollectiveOpsTpu.my_compiler
+    ) if use_dynamo else _dist_all_gather_into_tensor
+    f(output_tensor, input)
+    torch_xla.sync()
+    if not use_dynamo:
+      assert 'xla::AllGather' in met.counter_names(
+      ) or 'xla::AllGatherOut' in met.counter_names()
+    else:
+      assert 'xla::all_gather_into_tensor' in met.counter_names()
+    return output_tensor.cpu()
+
+  @staticmethod
+  def _all_gather(use_dynamo: bool):
+    met.clear_all()
+
+    def _dist_allgather(input):
```

Comment:
nit: make these inner functions have consistent naming patterns. Since they're used in such a limited context, you can also just given them all the same terse name like `callable`, `cc`, `program`, `model` etc.